rental_pricelist_timeunit/models/models.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, api
+
+class SaleOrderLineTU(models.Model):
+	_inherit = "sale.order.line"
+
+	def _get_display_price(self, product):
+		no_variant_attributes_price_extra = [
+			ptav.price_extra for ptav in self.product_no_variant_attribute_value_ids.filtered(
+				lambda ptav:
+					ptav.price_extra and
+					ptav not in product.product_template_attribute_value_ids
+			)
+		]
+		if no_variant_attributes_price_extra:
+			product = product.with_context(no_variant_attributes_price_extra=tuple(no_variant_attributes_price_extra))
+
+		if self.order_id.pricelist_id.discount_policy == 'with_discount':
+			if self.number_of_time_unit:
+				return product.with_context(pricelist=self.order_id.pricelist_id.id, uom=self.product_uom.id, quantity=self.number_of_time_unit).price
+			else:
+				return product.with_context(pricelist=self.order_id.pricelist_id.id, uom=self.product_uom.id).price
+
+		product_context = dict(self.env.context, partner_id=self.order_id.partner_id.id, date=self.order_id.date_order, uom=self.product_uom.id)
+
+		final_price, rule_id = self.order_id.pricelist_id.with_context(product_context).get_product_price_rule(product or self.product_id, self.product_uom_qty or 1.0, self.order_id.partner_id)
```

Comment:
Great job! Thanks for your contribution!
Can you please also do the improvement for the price without discount. Just calculate the final_price and base_price also in this way.
Can you please change the name file into sale_order_line.py and change the class name into SaleOrderLine

Reply:
this was a client requirement. 
all requested changes done, watch out for unintended consquences with final_price and base_price changes, but everything seems to work OK

Reply:
I've tested the feature "without discount". It doesn't work. The field "discount" of sale.order.line still depends on product_uom_qty. Maybe you should override the function _onchange_discount of sale.order.line

Reply:
I'll check this tomorrow, along with another bug I found with pricelists

Reply:
> I've tested the feature "without discount". It doesn't work. The field "discount" of sale.order.line still depends on product_uom_qty. Maybe you should override the function _onchange_discount of sale.order.line

#f34edb7 makes this work and the same time fixes malfunction using regular pricelists

Reply:
I think they belong to the same Feature. Can you please override the function _onchange_discount to make it also work for this Option. For me it is a bug if someone use the second Option and get a different price.
![Screenshot from 2023-11-22 13-41-53](https://github.com/elego/rental-vertical/assets/17932277/6a4d3251-01bd-45d3-8852-41aac9a23a91)

Reply:
Done, was done before, but duplicated